### PR TITLE
Tests: add tests for vcs.git.backend

### PR DESCRIPTION
**Pull request by indrajithi**
# Pull Request Check List

Related-to: [#3155](https://github.com/python-poetry/poetry/issues/3155)

- [X] Added **tests** for `poetry.vcs.git.backend`

### CODE REVIEW

File: tests/vcs/git/test_backend.py [hunk 1]
```diff
@@ -41,3 +47,25 @@ def test_invalid_revision_sha_max_len() -> None:
 def test_get_name_from_source_url(url: str) -> None:
     name = Git.get_name_from_source_url(url)
     assert name == "poetry"
+
+
+def test_annotated_tag() -> None:
```
Comment (radoering) [inline tests/vcs/git/test_backend.py:52]: Can you make the test `parametrized` to test passing `bytes` as well as `str`, please?

File: tests/vcs/git/test_backend.py [hunk 2]
```diff
@@ -41,3 +47,25 @@ def test_invalid_revision_sha_max_len() -> None:
...
+def test_get_remote_url() -> None:
+    repo = MagicMock(spec=Repo)
+    repo.get_config.return_value.get.return_value = (
+        b"https://github.com/python-poetry/poetry.git"
+    )
+
+    assert Git.get_remote_url(repo) == "https://github.com/python-poetry/poetry.git"
+
+
+def test_urlpathjoin() -> None:
```
Comment (radoering) [inline tests/vcs/git/test_backend.py:66]: Can you make the test `parametrized` to add an example with a trailing slash, e.g. `"ssh://git@github.com/org/repo/"`, please?

**Review by radoering — CHANGES_REQUESTED** [review]

File: tests/vcs/git/test_backend.py (hunk 1 above)
Comment (indrajithi) [reply tests/vcs/git/test_backend.py:52]: CMMIR, 
```
    @staticmethod
    def get_name_from_source_url(url: str) -> str:
        return re.sub(r"(.git)?$", "", url.rstrip("/").rsplit("/", 1)[-1])
```
 Does not accept bytes as url right? So passing bytes as url will fail unless we change this implementation.

File: tests/vcs/git/test_backend.py (hunk 2 above)
Comment (indrajithi) [reply tests/vcs/git/test_backend.py]: I have made the changes. LMK if I missed something or need to make any further changes. @radoering

**Review by radoering — APPROVED** [review]

**Pull request merged**